### ISSUE TITLE
Applying to a Job

### ISSUE THREAD

**Issue by erikaheidi** [AmsterdamPHP/job-board#37]
This comes as a suggestion for discussing, as we talked on irc. 

As a way of providing more statistics information for us and for the employers / recruiters, the "apply" mechanism should not point the user for a direct contact by email; instead, we should get the desired information from the user and, in some way, give it to the employer.

In order to make this a reality, we need: 
- some sort of identification from the user (LinkedIn and / or Github login for instance)
- saving user details in DB / storage (like User CV)
- saving application details in DB (who applied for that job, when, etc...)
- sending the application for the employer

The login leads to the possibility of having a user profile (we will have access to his info anyways).

This is not mandatory for going live with the project, but it will be really useful for the future.

**Comment by rdohms** [comment]
I like the idea of giving the user a "1 click apply" button to allow him to send his info from LinkedIn to the employer.
However:
1. it must be optional, we can't force people to have LinkedIn, GitHub or data in out own website.
2. I do not like the idea of keeping the user profile local. We already update so many sites, having to keep this one updated is not practical, if we do pull user data it should always be fresh from linkedin.

I think the best solution for both of these is to have a "Apply with Linkedin" button next to a regular apply button. I have seen another site use this, i think LinkedIn already has some nifty stuff for this, need to check.

that's my 2 cents.

**Comment by erikaheidi** [comment]
Well, fair enough...

So we could start with a form, what you think? A simple form with no auth required. Just to take care of the submission process.

Then, we can go for the "Apply with Linkedin" button.

This way we would have more reliable statistics. Basically, getting rid of the direct email behavior, that's what I mean.

**Comment by rdohms** [comment]
It would have the disadvantage that we would either require a user account or an email in order to set the 'from' address, and we wanted to avoid user accounts, leaving accounts only for people posting.

I'm not so much against the direct email approach, i would maybe label the button "Email something" or just leave a visible email address for contact. 

But i would like to see more opinions from the community who are our actual target market.

**Comment by rdohms** [comment]
BTW: https://developer.linkedin.com/apply-linkedin its a simple button.

**Comment by rdohms** [comment]
Yep, we can easily add this button, we already have all needed data, just need an API key and injecting the data into the template. Opening a ticket for this.

**Comment by erikaheidi** [comment]
Cool! I had no idea they already have this Apply button. Awesome

2013/5/1 Rafael Dohms notifications@github.com

**Comment by pascaldevink** [comment]
As mentioned in https://github.com/AmsterdamPHP/job-board/pull/40, it's a very ugly button. Besides that, it's not very flexible or easily extendible.
I'm more in favor of a simple form, again as mentioned in https://github.com/AmsterdamPHP/job-board/pull/40, which asks for an email address and a small text and in which people can enter links to their profiles on different services, like linkedin, github, bitbucket, mojolive, etc.

**Comment by rdohms** [comment]
As mentioned in #40 i think our own form and its dynamics are a downside to the user, linkedin is the most straight forward one, apart from simple click and email.

Also, I do not find the button that ugly at all... not more then bootstrap anyway.
